server/enhancedDate.js

```diff
@@ -1 +1,82 @@
-// export this as a Node module using the enhancedDate.js code from Module_2
\ No newline at end of file
+// export this as a Node module using the enhancedDate.js code from Module_2
+
+
+  
+ 'use strict';
+  
+  var theDate = {
+    time: new Date(),
+    days: ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 
+           'Friday', 'Saturday'],
+    months: ['January', 'February', 'March', 'April', 'May', 'June',
+             'July', 'August', 'September', 'October', 'November',
+             'December'],
+    
+  };
+  
+ 
+  
+    var setDate = function (d) {
+      
+      if (!(d instanceof Date)) {
+          theDate.time = new Date();
+      } else {
+        theDate.time = new Date(d);
+      }
+          return theDate.time;
+    };
+    
+    var getDate = function (param) {
```

Comment:
all functions besides setDate should check if a Date is set before trying to do things with theDate.time